src/Enemy.cpp

```diff
@@ -57,11 +57,12 @@ void Enemy::DisplayHUD() {
 
 
 
-vector<int> Enemy::GetDrops() {
+ItemMap_t Enemy::GetDrops() {
     // Returns the number of items dropped when the enemy is defeated depending on the item won.
-    vector<int> drops;
+    ItemMap_t drops;
     for (int i = 0; i < NUM_ITEMS; i++) {
-        drops.push_back(ReturnItemDrop(i));
+        ITEMTYPE itemType = static_cast<ITEMTYPE>(i + 1);
+        drops[itemType] =  ReturnItemDrop(i);
     }
     return drops;
 }
```

Comment:
Sorry, I may have miscommunicated - I was thinking of combining `GetDrops()` with `ReturnItemDrop(int)` since I see no reason why they're separate methods. 

```c++
map<ITEMTYPE, int> Enemy::GetDrops() {
    map<ITEMTYPE, int> drops;
    int ArrowSelector = Common::RandomInt(0, 5);
    int WhetstoneSelector = Common::RandomInt(0, 1);
    int PotionSelector = Common::RandomInt(0, 9);

    if (ArrowSelector <= 1) {
        drops[ITEMTYPE::ARROWS] = 0;
    } else if (ArrowSelector==2||ArrowSelector==3) {
        drops[ITEMTYPE::ARROWS] = 3;
    } else {
        drops[ITEMTYPE::ARROWS] = 5;
    }
    
    // ... etc ...

    return drops;
}
```

This way we can avoid the `static_cast` and explicitly generate the drops in one method.

Reply:
@tagniam done